Doc/library/os.path.rst

```diff
@@ -298,10 +298,10 @@ the :mod:`glob` module.)
 .. function:: join(path, *paths)
 
    Join one or more path components intelligently.  The return value is the
-   concatenation of *path* and any members of *\*paths* with exactly one
-   directory separator following each non-empty part except the last, meaning
-   that the result will only end in a separator if the last part is empty.  If
-   a component is an absolute path, all previous components are thrown away
+   concatenation of *path* and any non-empty members of *\*paths* separated
+   by the platform's directory separator character. If the last member of *\*paths*
+   is empty, the result will end in the platform's directory separator.
```

Comment:
Made this part a bit more specific, but can use original suggested wording if wanted

```suggestion
   is empty, the result will end in a directory separator.
```

Reply:
> If the last member of *\*paths*
> is empty, the result will end in the platform's directory separator.

It is not true. For example, `os.path.join('', '')` -> `''`. On Windows, `os.path.join('c:', '')` -> `'c:'`.

Reply:
That's a good point. Then perhaps instead, `is empty, and if *path* is not a root dir or empty, then ...`. It can also be optionally less specific, and say, `... the result will generally end in the ...`.

Reply:
It is problematic, because the term "root dir" needs explanation, especially on Windows.

What if just say that empty members are ignored, except the last one?

Reply:
If it needs some explanation, then maybe we give a quick example and phrase it as "... and if *path* is a root directory (i.e., / or c:) or empty, then ..."

For your second statement, do you mean leaving out the note on platform separators? It seems a bit too noteworthy to leave it out though. It could be phrased differently like,

If the last element of paths is empty, the result will end in the platform's directory separator, unless path is empty or a root directory.

Reply:
I afraid that it would be too verbose and still not clear. It is not good place to introduce a new term, especially if it may conflict with the meaning in other sites.

Reply:
I see, you would prefer to keep it a bit vague, like `The return value is the concatenation of *path* and any non-empty members of *paths* separated by the platform's directory separator character. Empty members of *paths* are ignored except for the last path. If a component is an absolute path, ...`?